index.js

```diff
@@ -65,39 +65,67 @@ var findFixmes = function(file){
   })
 }
 
-var eligibleFile = function(fp, excludePaths){
-  return (excludePaths.indexOf(fp.split("/code/")[1]) < 0) &&
-  !fs.lstatSync(fp).isDirectory() &&
-  (excludeExtensions.indexOf(path.extname(fp)) < 0)
+// Returns an array of unique array values not included in the other provided array
+var diff = function(a1, a2) {
+  var result = [];
+
+  for (var i = 0; i < a1.length; i++) {
+    if (a2.indexOf(a1[i]) === -1) {
+      result.push(a1[i]);
+    }
+
+  }
+  return result;
 }
 
-// Uses glob to traverse code directory and find files to analyze,
-// excluding files passed in with by CLI config
-var fileWalk = function(excludePaths){
-  var analysisFiles = [];
-  var allFiles = glob.sync("/code/**/**", {});
+// Returns all the file paths in the main directory that match the given pattern
+var buildFiles = function(paths) {
+  var files = [];
 
-  allFiles.forEach(function(file, i, a){
-    if(eligibleFile(file, excludePaths)){
-      analysisFiles.push(file);
-    }
+  paths.forEach(function(path, i, a) {
+    var pattern = "/code/" + path + "**"
+    files.push.apply(files, glob.sync(pattern, {}));
+  });
+
+  return files;
+}
+
+// Filters the directory paths out
+var filterFiles = function(files) {
+  return files.filter(function(file) {
+    return !fs.lstatSync(file).isDirectory();
```

Comment:
Is there an `isFile` we can use to avoid symlinks? Or am I wrong about our policy on symlinks?